artipie-main/src/test/java/com/artipie/conda/CondaS3ITCase.java

```diff
@@ -0,0 +1,340 @@
+import java.io.IOException;
+import org.hamcrest.Matchers;
+import org.hamcrest.core.IsEqual;
+import org.hamcrest.core.StringContains;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.EnabledOnOs;
+import org.junit.jupiter.api.condition.OS;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+import org.testcontainers.containers.BindMode;
+import org.testcontainers.containers.wait.strategy.AbstractWaitStrategy;
+
+/**
+ * Conda IT case with S3 storage.
+ * @since 0.23
+ * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
+ * @checkstyle LineLengthCheck (500 lines)
+ */
+@EnabledOnOs({OS.LINUX, OS.MAC})
+@SuppressWarnings("PMD.AvoidDuplicateLiterals")
+public final class CondaS3ITCase {
+
+    /**
+     * Curl exit code when resource not retrieved and `--fail` is used, http 400+.
+     */
+    private static final int CURL_NOT_FOUND = 22;
+
+    /**
+     * Repository TCP port number.
+     */
+    private static final String PORT = "8080";
+
+    /**
+     * Test repository name.
+     */
+    private static final String REPO = "my-conda";
+
+    /**
+     * Test deployments.
+     * @checkstyle VisibilityModifierCheck (10 lines)
+     * @checkstyle MagicNumberCheck (10 lines)
+     */
+    @RegisterExtension
+    final TestDeployment containers = new TestDeployment(
+        () -> TestDeployment.ArtipieContainer.defaultDefinition()
+            .withUser("security/users/alice.yaml", "alice")
+            .withRepoConfig("conda/conda-s3.yml", "my-conda"),
+        () -> new TestDeployment.ClientContainer("continuumio/miniconda3:4.10.3")
+            .withWorkingDirectory("/w")
+            .withNetworkAliases("minic")
+            .withExposedPorts(9000)
+            .waitingFor(
+                new AbstractWaitStrategy() {
+                    @Override
+                    protected void waitUntilReady() {
+                        // Don't wait for minIO port.
+                    }
+                }
+            )
+            .withClasspathResourceMapping(
+                "conda/condarc", "/w/.condarc", BindMode.READ_ONLY
+            )
+            .withClasspathResourceMapping(
+                "conda/snappy-1.1.3-0.tar.bz2", "/w/snappy-1.1.3-0.tar.bz2", BindMode.READ_ONLY
+            )
+            .withClasspathResourceMapping(
+                "conda/noarch_glom-22.1.0.tar.bz2", "/w/noarch_glom-22.1.0.tar.bz2", BindMode.READ_ONLY
+            )
+            .withClasspathResourceMapping(
+                "conda/linux-64_nng-1.4.0.tar.bz2", "/w/linux-64_nng-1.4.0.tar.bz2", BindMode.READ_ONLY
+            )
+            .withClasspathResourceMapping(
+                "minio-bin-20231120.txz", "/w/minio-bin-20231120.txz", BindMode.READ_ONLY
+            )
+    );
+
+    @BeforeEach
+    void init() throws IOException {
```

Comment:
Maybe we can make Dockerfile? As we discussed it's it will faster with way and it's simple with `TestContainers`, check [the docs](https://java.testcontainers.org/features/creating_images/) and example:
```
new GenericContainer<>(
    new ImageFromDockerfile("maven-client").withFileFromClasspath(
        "Dockerfile", "ArtipieAndPluginFromCentralIT/Dockerfile"
    )
)
```
`maven-client` is container name, `ArtipieAndPluginFromCentralIT/Dockerfile` is test Dockerfile from recourses.